### ISSUE TITLE
Flaky test: KafkaSingleMessageReaderTest shouldFetchSingleAvroMessage and shouldFetchSingleAvroMessageWithSchemaAwareSerialization

### ISSUE THREAD

**Issue by pitagoras3** [allegro/hermes#1454]
Both tests (_shouldFetchSingleAvroMessage_ and _shouldFetchSingleAvroMessageWithSchemaAwareSerialization_) are failing due to possible 201 instead of expected 202:
```
2022-02-02T12:59:01.7251097Z Gradle suite > Gradle test > pl.allegro.tech.hermes.integration.KafkaSingleMessageReaderTest > shouldFetchSingleAvroMessage SKIPPED
2022-02-02T12:59:02.5246748Z 
2022-02-02T12:59:02.5260766Z Gradle suite > Gradle test > pl.allegro.tech.hermes.integration.KafkaSingleMessageReaderTest > shouldFetchSingleAvroMessage SKIPPED
2022-02-02T12:59:03.8246067Z 
2022-02-02T12:59:03.8247207Z Gradle suite > Gradle test > pl.allegro.tech.hermes.integration.KafkaSingleMessageReaderTest > shouldFetchSingleAvroMessage FAILED
2022-02-02T12:59:03.8247908Z     org.junit.ComparisonFailure: expected:<20[1]> but was:<20[2]>
2022-02-02T12:59:03.8248479Z         at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2022-02-02T12:59:03.8249224Z         at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2022-02-02T12:59:03.8250025Z         at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2022-02-02T12:59:03.8250857Z         at pl.allegro.tech.hermes.integration.test.HttpResponseAssertion.hasStatus(HttpResponseAssertion.java:19)
2022-02-02T12:59:03.8252070Z         at pl.allegro.tech.hermes.integration.KafkaSingleMessageReaderTest.shouldFetchSingleAvroMessage(KafkaSingleMessageReaderTest.java:74)
```

**Comment by sobelek** [comment]
Hey. @pitagoras3 
According to this line hermes should always return 201 if request was sucessfull. How is it possible that there is also a 202? Is it expected behavior? https://github.com/allegro/hermes/blob/7340b364c2f82642487545026bb95f2865feeb90/hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/TopicsEndpoint.java#L103

**Comment by pitagoras3** [comment]
Hi @sobelek,
Hermes can return *202 (Accepted)* HTTP status code, when event published to Hermes has not been acknowledged by Kafka - in such case Hermes is [buffering](https://hermes-pubsub.readthedocs.io/en/latest/user/publishing/#buffering) it and will try to deliver it ASAP. This behaviour is described in [documentation](https://hermes-pubsub.readthedocs.io/en/latest/user/publishing/#response-codes).

**Comment by sobelek** [comment]
Alright, so this is actually a correct behavior and test should be ready to get 201 or 202. I will fix it and try to find if any other test should handle this behavior.

**Comment by sobelek** [comment]
This behavior was actually fixed here: 5c07600bcad5dd9f681237fef83b93ebfc869839

**Comment by pitagoras3** [comment]
Yep, looks like @szczygiel-m fixed it in https://github.com/allegro/hermes/pull/1476, so I'm closing this issue :)
